app/views/project/editProject.js

```diff
@@ -57,11 +58,47 @@ let editProject = (cursor) => {
       })
 }
 
+let DeleteProjectDialog = component('DeleteProjectDialog', (cursor) => {
+  
+  let yesCallback = () => {
+    let editCursor = cursor.cursor('editProject')
+    let projectCursor = editCursor.cursor('project')
+    let project = projectCursor.toJS()
+    let qualifiedProjectId = `${project.owner}/${project.projectId}`
+    
+    editCursor.set('isWaiting', true)
+    
+    ajax.delete(`/api/projects/${project.owner}/${project.projectId}`)
+      .then(() => {
+        logger.debug(`Project successfully deleted '${qualifiedProjectId}'`)
+        router.goTo('/')
+      }, (error) => {
+        logger.warn(`Failed to delete project '${qualifiedProjectId}': ${error}`)
+      })
+      .finally(() => {
+        cursor.set('showDeleteProjectDialog', false)
+        editCursor.set('isWaiting', false)
+      })
+  }
+
+  let closeCallback = () => {
+    cursor.set('showDeleteProjectDialog', false)
+  }
+  let title = 'Delete'
+  let message = 'Are you sure you want to delete this project?'
+
+  return notification.question(title, message, yesCallback, closeCallback)
+  
```

Comment:
I tend to like compact code, there is a difference between unnecessary whitespace and delimiting whitespace IMO. By the latter I mean whitespace that delimits logical blocks of code. Whitespace at the beginning and especially the end of functions just read like noise to me. I always ask myself whether that was left on purpose or not, because it just looks like it was left behind by accident.